### PR TITLE
[Merged by Bors] - refactor(AlgebraicGeometry/EllipticCurve/*): rename baseChange to map

### PR DESCRIPTION
**Pull request by Multramate**
This is in accordance with other similar definitions e.g. `Polynomial.map`. It turns out that rewrite lemmas of the form `(W.map (φ.comp ψ)).negY/addX/addY = ψ ((W.map φ).negY/addX/addY)` are rather annoying to use, so they are replaced with the original `baseChange` functionality that says `(W.baseChange B).negY/addX/addY = ψ ((W.baseChange A).negY/addX/addY)`. This addresses the issue in #9596 pointed out by @riccardobrasca, but for the Weierstrass curve `W` rather than its points.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean [hunk 1]
```diff
@@ -780,170 +780,199 @@ end Group
...
 
-/-! ### Base changes -/
+/-! ### Maps and base changes -/
 
-variable {A : Type v} [CommRing A] (φ : R →+* A) {B : Type w} [CommRing B] (ψ : A →+* B)
+variable {A : Type v} [CommRing A] (φ : R →+* A)
 
-lemma equation_iff_baseChange {φ : R →+* A} (hφ : Function.Injective φ) (x y : R) :
-    W.equation x y ↔ (W.baseChange φ).toAffine.equation (φ x) (φ y) := by
+lemma map_equation {φ : R →+* A} (hφ : Function.Injective φ) (x y : R) :
```
Comment (riccardobrasca) [inline Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean:787]: Shouldn't the name still contain `iff`? Also below.

File: Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean [hunk 2]
```diff
@@ -17,8 +17,8 @@ under the geometric group law defined in `Mathlib.AlgebraicGeometry.EllipticCurv
 
 Let `W` be a Weierstrass curve over a field `F` given by a Weierstrass equation $W(X, Y) = 0$ in
 affine coordinates. As in `Mathlib.AlgebraicGeometry.EllipticCurve.Affine`, the set of nonsingular
-rational points $W(F)$ of `W` consist of the unique point at infinity $0$ and nonsingular affine
-points $(x, y)$. With this description, there is an addition-preserving injection between $W(F)$
+rational points $W⟮F⟯$ of `W` consist of the unique point at infinity $0$ and nonsingular affine
```
Comment (riccardobrasca) [inline Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean:20]: Those are the weird parenthesis used for for `IntermediateField` (or something similar), right? I guess LaTeX will not love them

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean (hunk 2 above)
Comment (Multramate) [reply Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean:20]: Sorry, thanks!

File: Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean (hunk 1 above)
Comment (Multramate) [reply Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean:787]: Eh, I think this is probably fine.

**Pull request closed without merging**